src/main/java/seedu/student/logic/commands/EditAppointmentCommand.java

```diff
@@ -0,0 +1,187 @@
+import seedu.student.model.appointment.SameDateAppointmentList;
+import seedu.student.model.student.MatriculationNumber;
+
+public class EditAppointmentCommand extends Command {
+    public static final String COMMAND_WORD = "editAppt";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits an appointment identified in Vax@NUS \n"
+            + "by the index number used in the displayed student list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: Matriculation Number "
+            + PREFIX_MATRICULATION_NUMBER + "MATRICULATION NUMBER "
+            + "[" + PREFIX_DATE + "DATE] "
+            + "[" + PREFIX_START_TIME + " START TIME] \n"
+            + "Example: " + PREFIX_MATRICULATION_NUMBER + " A01234567X "
+            + PREFIX_DATE + "2021-03-25 "
+            + PREFIX_START_TIME + "11:00 \n";
+
+    public static final String MESSAGE_EDIT_APPOINTMENT_SUCCESS = "Edited Appointment: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_APPOINTMENT = "The appointment overlaps with existing records";
+    public static final String MESSAGE_APPOINTMENT_DOES_NOT_EXIST = "The requested appointment does not exist";
+
+    private final MatriculationNumber matriculationNumber;
+    private final EditAppointmentDescriptor editAppointmentDescriptor;
+
+    /**
+     * @param matriculationNumber of the student in the appointment in the filtered appointment list to edit
+     * @param editAppointmentDescriptor details to edit the appointment with
+     */
+    public EditAppointmentCommand(MatriculationNumber matriculationNumber,
+                                  EditAppointmentDescriptor editAppointmentDescriptor) {
+        requireNonNull(matriculationNumber);
+        requireNonNull(editAppointmentDescriptor);
+
+        this.matriculationNumber = matriculationNumber;
+        this.editAppointmentDescriptor = new EditAppointmentDescriptor(editAppointmentDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+
+        Appointment appointmentToEdit = null;
+        List<SameDateAppointmentList> lastShownList = model.getStudentBook().getAppointmentList();
+        for (SameDateAppointmentList sList : lastShownList) {
+            for (Appointment a : sList) {
+                if (a.getMatriculationNumber().equals(matriculationNumber)) {
+                    appointmentToEdit = a;
+                }
+            }
+        }
+        if (appointmentToEdit == null) {
+            throw new CommandException(MESSAGE_APPOINTMENT_DOES_NOT_EXIST);
+        }
+
+        Appointment editedAppointment = createEditedAppointment(appointmentToEdit, editAppointmentDescriptor);
+
+        boolean studentExists = model.isExistingMatricNumber(editedAppointment.getMatriculationNumber());
```

Comment:
Is the check for whether the student exists necessary, as the existence of the appointment implies that the student must exist? Or do you consider this as defensive programming? Does it mean if the `Model` happens to have an appointment that corresponds to no student, this appointment cannot be edited?

Reply:
> Or do you consider this as defensive programming?
Yes, this is defensive. 

>  Does it mean if the Model happens to have an appointment that corresponds to no student, this appointment cannot be edited
Yes